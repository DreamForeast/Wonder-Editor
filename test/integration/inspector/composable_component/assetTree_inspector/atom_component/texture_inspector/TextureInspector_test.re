open Wonder_jest;

open Expect;

open Expect.Operators;

open AssetNodeType;

open Sinon;

let _ =
  describe("TextureInspector", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initStateAndGl(~sandbox, ());
      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => {
      restoreSandbox(refJsObjToSandbox(sandbox^));
      StateAssetService.getState()
      |> CurrentNodeDataAssetService.clearCurrentNodeData
      |> CurrentNodeParentIdAssetService.clearCurrentNodeParentId
      |> StateAssetService.setState
      |> ignore;
    });

    describe("prepare currentSelectSource", () => {
      beforeEach(() => {
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorAssetTool.initAssetTree,
        );
        CurrentSelectSourceEditorService.setCurrentSelectSource(
          EditorType.Asset,
        )
        |> StateLogicService.getAndSetEditorState;
      });

      describe("test component snapshot", () =>
        test("test texture inspector->show default value", () => {
          let assetTreeDomRecord =
            MainEditorAssetTool.buildTwoLayerAssetTreeRoot();

          assetTreeDomRecord
          |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
          |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;

          BuildComponentTool.buildInspectorComponent(
            TestTool.buildEmptyAppState(),
            InspectorTool.buildFakeAllShowComponentConfig(),
          )
          |> ReactTestTool.createSnapshotAndMatch;
        })
      );

      describe("test texture rename", () => {
        describe("test snapshot", () =>
          test("test rename to specific name", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let newName = "newTextureName";
            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;

            TextureInspectorTool.triggerInspectorRenameEvent(newName);

            BuildComponentTool.buildAssetComponent()
            |> ReactTestTool.createSnapshotAndMatch;
          })
        );

        describe("test logic", () =>
          describe("test engine", () => {
            beforeEach(() => {
              MainEditorAssetTool.buildFakeFileReader();
              MainEditorAssetTool.buildFakeImage();
            });
            testPromise("upload texture;
              rename texture;", () => {
              let assetTreeDomRecord =
                MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
              let newName = "newTextureToEngine";

              MainEditorAssetHeader.Method._fileLoad(
                TestTool.getDispatch(),
                BaseEventTool.buildFileEvent(),
              )
              |> Js.Promise.then_(() => {
                   assetTreeDomRecord
                   |> MainEditorAssetNodeTool.OperateTwoLayer.getUploadedeTextureNodeDomIndex
                   |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;
                   TextureInspectorTool.triggerInspectorRenameEvent(newName);

                   MainEditorAssetNodeTool.getTextureIndexFromCurrentNodeId()
                   |> BasicSourceTextureEngineService.unsafeGetBasicSourceTextureName
                   |> StateLogicService.getEngineStateToGetData
                   |> expect == newName
                   |> Js.Promise.resolve;
                 });
            });
          })
        );
      });

      describe("test texture change wrap", () => {
        describe("test set wrapS to REPEAT", () => {
          test("test snapshot", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let wrapSDomIndex = TextureInspectorTool.getWrapSDomIndex();
            let wrapRepeatType = TextureInspectorTool.getWrapRepeatType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;
            TextureInspectorTool.triggerInspectorChangeWrapEvent(
              wrapSDomIndex,
              wrapRepeatType,
            );

            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });
          test("test logic", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let wrapSDomIndex = TextureInspectorTool.getWrapSDomIndex();
            let wrapRepeatType = TextureInspectorTool.getWrapRepeatType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;

            TextureInspectorTool.triggerInspectorChangeWrapEvent(
              wrapSDomIndex,
              wrapRepeatType,
            );

            let textureIndex =
              TextureInspectorTool.getTextureIndexFromCurrentNodeData();

            BasicSourceTextureEngineService.getWrapS(textureIndex)
            |> StateLogicService.getEngineStateToGetData
            |> TextureTypeUtils.convertWrapToInt
            |> expect == wrapRepeatType;
          });
        });

        describe("test set wrapT to MIRRORED_REPEAT", () => {
          test("test snapshot", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let wrapTDomIndex = TextureInspectorTool.getWrapTDomIndex();
            let wrapMirroredRepeatType =
              TextureInspectorTool.getWrapMirroredRepeatType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;

            TextureInspectorTool.triggerInspectorChangeWrapEvent(
              wrapTDomIndex,
              wrapMirroredRepeatType,
            );

            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });
          test("test logic", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let wrapTDomIndex = TextureInspectorTool.getWrapTDomIndex();
            let wrapMirroredRepeatType =
              TextureInspectorTool.getWrapMirroredRepeatType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;
            TextureInspectorTool.triggerInspectorChangeWrapEvent(
              wrapTDomIndex,
              wrapMirroredRepeatType,
            );

            let textureIndex =
              TextureInspectorTool.getTextureIndexFromCurrentNodeData();

            BasicSourceTextureEngineService.getWrapT(textureIndex)
            |> StateLogicService.getEngineStateToGetData
            |> TextureTypeUtils.convertWrapToInt
            |> expect == wrapMirroredRepeatType;
          });
        });
      });

      describe("test texture change filter", () => {
        describe("test set MagFilter to LINEAR_MIPMAP_LINEAR", () => {
          test("test snapshot", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let magFilterDomIndex =
              TextureInspectorTool.getMagFilterDomIndex();
            let filterLinearMipmapLinearType =
              TextureInspectorTool.getFilterLinearMipmapLinearType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;

            TextureInspectorTool.triggerInspectorChangeFilterEvent(
              magFilterDomIndex,
              filterLinearMipmapLinearType,
            );

            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });
          test("test logic", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let magFilterDomIndex =
              TextureInspectorTool.getMagFilterDomIndex();
            let filterLinearMipmapLinearType =
              TextureInspectorTool.getFilterLinearMipmapLinearType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;
            TextureInspectorTool.triggerInspectorChangeFilterEvent(
              magFilterDomIndex,
              filterLinearMipmapLinearType,
            );

            let textureIndex =
              TextureInspectorTool.getTextureIndexFromCurrentNodeData();

            BasicSourceTextureEngineService.getMagFilter(textureIndex)
            |> StateLogicService.getEngineStateToGetData
            |> TextureTypeUtils.convertFilterToInt
            |> expect == filterLinearMipmapLinearType;
          });
        });

        describe("test set MinFilter to NEAREST_MIPMAP_LINEAR", () => {
          test("test snapshot", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let minFilterDomIndex =
              TextureInspectorTool.getMinFilterDomIndex();
            let filterNearestMipmapLinearType =
              TextureInspectorTool.getFilterNearestMipmapLinearType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;
            TextureInspectorTool.triggerInspectorChangeFilterEvent(
              minFilterDomIndex,
              filterNearestMipmapLinearType,
            );

            BuildComponentTool.buildInspectorComponent(
              TestTool.buildEmptyAppState(),
              InspectorTool.buildFakeAllShowComponentConfig(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });

          test("test logic", () => {
            let assetTreeDomRecord =
              MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
            let minFilterDomIndex =
              TextureInspectorTool.getMinFilterDomIndex();
            let filterNearestMipmapLinearType =
              TextureInspectorTool.getFilterNearestMipmapLinearType();

            assetTreeDomRecord
            |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstTextureDomIndex
            |> MainEditorAssetTool.clickAssetChildrenNodeToSetCurrentNode;
            TextureInspectorTool.triggerInspectorChangeFilterEvent(
              minFilterDomIndex,
              filterNearestMipmapLinearType,
            );

            let textureIndex =
              TextureInspectorTool.getTextureIndexFromCurrentNodeData();

            BasicSourceTextureEngineService.getMinFilter(textureIndex)
            |> StateLogicService.getEngineStateToGetData
            |> TextureTypeUtils.convertFilterToInt
            |> expect == filterNearestMipmapLinearType;
          });
        });
      });
    });
  });