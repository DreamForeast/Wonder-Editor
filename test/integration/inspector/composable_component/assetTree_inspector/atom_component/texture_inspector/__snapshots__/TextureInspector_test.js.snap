// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextureInspector prepare currentSelectSource test component snapshot test texture inspector->show default value 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <div
        className=""
      >
        <h1>
          Texture
        </h1>
        <hr />
        <div
          className=""
        >
          <article
            className="wonder-string-input"
          >
            <span
              className="component-label"
            >
              name : 
            </span>
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="texture3"
            />
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapS Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapT Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Mag Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Min Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`TextureInspector prepare currentSelectSource test texture change filter test set MagFilter to LINEAR_MIPMAP_LINEAR test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <div
        className=""
      >
        <h1>
          Texture
        </h1>
        <hr />
        <div
          className=""
        >
          <article
            className="wonder-string-input"
          >
            <span
              className="component-label"
            >
              name : 
            </span>
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="texture3"
            />
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapS Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapT Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Mag Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="5"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Min Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`TextureInspector prepare currentSelectSource test texture change filter test set MinFilter to NEAREST_MIPMAP_LINEAR test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <div
        className=""
      >
        <h1>
          Texture
        </h1>
        <hr />
        <div
          className=""
        >
          <article
            className="wonder-string-input"
          >
            <span
              className="component-label"
            >
              name : 
            </span>
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="texture3"
            />
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapS Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapT Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Mag Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Min Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="4"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`TextureInspector prepare currentSelectSource test texture change wrap test set wrapS to REPEAT test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <div
        className=""
      >
        <h1>
          Texture
        </h1>
        <hr />
        <div
          className=""
        >
          <article
            className="wonder-string-input"
          >
            <span
              className="component-label"
            >
              name : 
            </span>
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="texture3"
            />
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapS Mode
            </span>
            <select
              onChange={[Function]}
              value="2"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapT Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Mag Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Min Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`TextureInspector prepare currentSelectSource test texture change wrap test set wrapT to MIRRORED_REPEAT test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <div
        className=""
      >
        <h1>
          Texture
        </h1>
        <hr />
        <div
          className=""
        >
          <article
            className="wonder-string-input"
          >
            <span
              className="component-label"
            >
              name : 
            </span>
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="texture3"
            />
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapS Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapT Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Mag Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Min Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`TextureInspector prepare currentSelectSource test texture rename test snapshot test rename to specific name 1`] = `
<article
  className="wonder-asset-component"
>
  <div
    className="asset-tree"
  >
    <article
      className="wonder-asset-header"
    >
      <div
        className="header-item"
      >
        <button
          onClick={[Function]}
        >
          addFolder
        </button>
      </div>
      <div
        className="header-item"
      >
        <button
          disabled={false}
          onClick={[Function]}
        >
          remove
        </button>
      </div>
      <div
        className="header-item"
      >
        <input
          className="file-upload"
          multiple={true}
          onChange={[Function]}
          type="file"
        />
      </div>
    </article>
    <article
      className="wonder-asset-assetTree"
    >
      <ul
        className="wonder-tree-node"
      >
        <li
          onClick={[Function]}
          style={
            Object {
              "background": "#c0c0c0",
            }
          }
        >
          <div
            className="item-ground"
            draggable={true}
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
          />
          <img
            src="./public/img/12.jpg"
          />
          Asset
        </li>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid red",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <img
              src="./public/img/12.jpg"
            />
            newFolder
          </li>
        </ul>
        <ul
          className="wonder-tree-node"
          draggable={true}
          onDragEnd={[Function]}
          onDragStart={[Function]}
        >
          <li
            onClick={[Function]}
            style={
              Object {
                "border": "1px solid red",
              }
            }
          >
            <div
              className="item-ground"
              draggable={true}
              onDragEnter={[Function]}
              onDragLeave={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            />
            <img
              src="./public/img/12.jpg"
            />
            newFolder
          </li>
        </ul>
      </ul>
    </article>
  </div>
  <article
    className="wonder-asset-assetChildren"
  >
    <article
      className="wonder-asset-folderBox"
      id="folder-2"
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <img
        src="./public/img/11.jpg"
      />
      <span
        className="item-text"
      >
        newFolder
      </span>
    </article>
    <article
      className="wonder-asset-folderBox"
      id="folder-3"
      style={
        Object {
          "border": "1px solid red",
        }
      }
    >
      <div
        className="item-ground"
        draggable={true}
        onDragEnd={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
      />
      <img
        src="./public/img/11.jpg"
      />
      <span
        className="item-text"
      >
        newFolder
      </span>
    </article>
    <article
      className="wonder-asset-fileBox item-active"
      onClick={[Function]}
    >
      <img
        onDragStart={[Function]}
        src="texture3img"
      />
      <span
        className="item-text"
      >
        newTextureName
      </span>
    </article>
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <img
        onDragStart={[Function]}
        src="./public/img/12.jpg"
      />
      <span
        className="item-text"
      >
        json.json
      </span>
    </article>
    <article
      className="wonder-asset-fileBox "
      onClick={[Function]}
    >
      <img
        onDragStart={[Function]}
        src="texture5img"
      />
      <span
        className="item-text"
      >
        texture5
      </span>
    </article>
  </article>
</article>
`;
